tardis/montecarlo/montecarlo_numba/tests/test_packet.py

```diff
@@ -0,0 +1,303 @@
+import os
+import pytest
+import numpy as np
+import pandas as pd
+import tardis.montecarlo.formal_integral as formal_integral
+import tardis.montecarlo.montecarlo_numba.r_packet as r_packet
+import tardis.montecarlo.montecarlo_configuration as mc
+import tardis.montecarlo.montecarlo_numba.numba_interface as numba_interface
+from tardis import constants as const
+from tardis.montecarlo.montecarlo_numba.numba_interface import Estimators
+import tardis.montecarlo.montecarlo_numba.numba_config as numba_config
+from tardis.montecarlo.montecarlo_numba import macro_atom
+C_SPEED_OF_LIGHT = const.c.to('cm/s').value
+
+from numpy.testing import (
+        assert_equal,
+        assert_almost_equal,
+        assert_array_equal,
+        assert_allclose
+        )
+
+@pytest.fixture(scope="function")
+def packet():
+    return r_packet.RPacket(
+        r = 7.5e14,
+        nu = 0.4,
+        mu = 0.3,
+        energy = 0.9,
+        seed = 1963,
+        index = 0,
+        is_close_line = 0
+    )
+
+@pytest.fixture(scope="function")
+def model():
+    return numba_interface.NumbaModel(
+        r_inner = np.array([6.912e14, 8.64e14], dtype=np.float64),
+        r_outer = np.array([8.64e14, 1.0368e15], dtype=np.float64),
+        time_explosion = 5.2e7
+    )
+
+@pytest.fixture(scope="function")
+def estimators():
+    return numba_interface.Estimators(
+        j_estimator = np.array([0.0, 0.0], dtype=np.float64),
+        nu_bar_estimator = np.array([0.0, 0.0], dtype=np.float64),
+        j_blue_estimator = np.array([[0.0, 0.0, 0.0], [0.0, 0.0, 0.0]], dtype=np.float64),
+        Edotlu_estimator = np.array([[0.0, 0.0, 1.0], [0.0, 0.0, 1.0]], dtype=np.float64)
+    )
+
+@pytest.mark.parametrize(
+    ['packet_params', 'expected_params'],
+    [({'mu': 0.3, 'r': 7.5e14},
+      {'d_boundary': 259376919351035.88}),
+
+     ({'mu': -.3, 'r': 7.5e13},
+      {'d_boundary': -664987228972291.5}),
+
+     ({'mu': -.3, 'r': 7.5e14},
+      {'d_boundary': 709376919351035.9})]
+)
```

Comment:
@andrewfullard @Rodot- we should use more easy to understand mus and nus here. I would say put it in the middle of the cell and go mu=1, -1 - and then mu=0 and use the secant calculation to see if thins calculates the thing. Does it also not return if it goes out or in (`d_shell`)?